### DISCUSSION TITLE
[DISCUSSION] When does a mentor get "paid"?

### DISCUSSION THREAD

**Discussion by iHiD** [exercism/v3#1467] [General]
We're thinking through **when** mentors acquire their credits and reputation.

- Student puts a bounty on a solution. 
- Bounty goes into escrow. 
- Mentor picks up the solution and provides some feedback.

We then have the following scenarios:
1. Both parties are satisfied
2. Student times out (no response after X days)
3. Mentor times out (no response after X days)
4. Mentor is helpful but student isn’t satisfied 
5. Mentor is unhelpful

- For 1, they both tick a button and the credit/reputation is awarded.
- For 2 or 3 it's awarded or refunded on timeout.
- 4 and 5 the problematic ones. In these "conflicted" situations how do we decide whether the mentor should get the token or the student get a refund? This is also with an extract contextual risk of one party trying to game the system. How do we disincentive these scenarios, while ensuring things stay fair?

Thoughts?

**Comment by bobrundle** [comment]
Well.  Not sure this will work.  Assumes good behavior of both parties.  Enables scenario where mentor and student are same person logged through separate accounts.  Student continuously creates bounties which mentor collects.  This can't happen on Stack Overflow because the person putting a bounty on a problem has to give up that much reputation to do it.  Here are you simply creating "money" out of nothing.  This is a basic conservation of reputation problem.

**Comment by iHiD** [comment]
> . Here are you simply creating "money" out of nothing. 

You can only spend credits you have, so I don't think this is a problem.

> Assumes good behavior of both parties.

It does, as does most of Exercism, and open source :) So the question, is how we incentivise that good behaviour, and/or deincentivise bad behaviour.

**Comment by neenjaw** [comment]
Agreed for situations 1-3.

### Situation 5

Situation 5 seems more straight forward than 4 -- Maybe there needs to be some sort of resolution process that requires the student to fill in some information as to _why_ they believe the mentor was unhelpful, maybe pre-populated options and a text box? But then that creates some onus for another system to then step in to resolve those problems.

Then maybe this could be a slider that allows for partial refund/reward with some comment to both parties?

I think the risk is that making this option very visible is that you might have people with petty disagreements have difficulty looking past themselves and filing a frivolous complaint, but too hidden then you might have people that give up with real issues and abandon the platform.

Is there a thought to developing some "Introduction to the mentoring experience" for students, which outlines the roles of both parties and lays some "ground rules" for the interaction? e.g. [this](https://www.npd-solutions.com/groundrules.html)

### Situation 4

I think this is probably the most stressful situation for the mentor because it can probably be sensed as the interaction takes place before the bounty is settled.  Probably the first thing is to be proactive and institute something like the ground rules mentioned above to guide expectations of both parties.

Whether there is a slider for some partial award/refund I think is the lesser issue at hand, because if the mentor is helpful and the student is sabotaging the effort the mentor shoul dhave some reward for their time.  The bigger issue is how to retain the mentor in a difficult situation that is outside of their control and how to help the student to realise their behaviour and realign them to create better interactions in the future.

**Comment by coriolinus** [comment]
The most straightforward system for dealing with cases 4 and 5 is to have another human judge. At the conclusion of mentoring, the student's options are "award bounty" or "go to arbitration". The arbiters are just other mentors, which raises collusion possibilities which I don't think are worth worrying too much about until they appear. Arbiters then get to choose from three options: award bounty to mentor, refund bounty to student, burn bounty (in case both parties were misbehaving).

**Comment by iHiD** [comment]
@coriolinus Nice. Yeah. 

And 50% of the bounty goes to the arbiter (regardless of the decision) to discourage this and encourage the arbiter...

**Comment by coriolinus** [comment]
I thought about awarding part of the payment to the arbiter regardless, but
then there's the problem that a malicious student can take a good mentor to
arbitration just to reduce their commission. It might be better to pay
arbiters in a distinct currency.

**Comment by taiyab** [comment]
I think you severely reduce malpractice by making the currency not translate to any **major** material rewards IRL.

**Comment by iHiD** [comment]
The "distinct currency" would probably be reputation here.

**Comment by workingjubilee** [comment]
I feel like I should note that cancellation fees for contracts exist.

In general it actually helps in terms of the (*sigh*) game theory if both parties "ante up" a token in order to manage transaction overhead. It can be the same ante, it can be different, but the catch is that if _both_ parties are in, and then one side breaks faith, then you have e.g. the option of allowing one party to go all "from Hell's heart, I stab at thee!" and burn the _ante_ for both participants. But if they reach a mutual agreement dissolution, then they keep their ante.

Then they can _recoup damages_ in arbitrage, and generating arbitrage work isn't free, so the "malicious student takes mentor to court just to try to get a better markup" is disincentivized.
